client/components/settings/local_settings_modal_controller.js

```diff
@@ -0,0 +1,41 @@
+class LocalSettingsModalController {
+  constructor (localSettings) {
+    Object.assign(this, { localSettings })
+    this.settings = {}
+  }
+
+  $onInit () {
+    this.loadSettings()
+  }
+
+  loadSettings () {
+    this.settings = this.localSettings.getFromLocalStorage()
+    this.settingsData = {}
+    // settingsData makes it much easier to properly display the settings in the modal itself
+    Object.keys(this.settings).forEach(settingsKey => {
+      const namespace = settingsKey.split('-')[0]
+      const data = { name: settingsKey.split('-')[1], value: this.settings[settingsKey].value, dataType: this.settings[settingsKey].dataType }
+      if (!this.settingsData.hasOwnProperty(namespace)) {
+        this.settingsData[namespace] = []
+        this.settingsData[namespace].push(data)
+      } else {
```

Comment:
Remove this else and take the line `this.settingsData[namespace].push(data)` out of the condition.